spec/system/admin/bulk_url_upload_spec.rb

```diff
@@ -1,27 +1,68 @@
 # frozen_string_literal: true
 
-describe 'Bulk URL upload' do
-  include ActiveJob::TestHelper
+shared_examples 'a successful bulk upload' do
+  it 'sends us back to the bulk upload page' do
+    do_bulk_upload
+    expect(page).to have_text('Bulk Search.gov URL Upload')
+  end
+
+  it 'shows a confirmation message' do
+    do_bulk_upload
+    expect(page).to have_text(
+      <<~CONFIRMATION_MESSAGE
+        Successfully uploaded #{upload_filename} for processing.
+        The results will be emailed to you.
+      CONFIRMATION_MESSAGE
+    )
+  end
 
-  subject(:bulk_upload) do
-    perform_enqueued_jobs do
-      visit url
-      attach_file('bulk_upload_urls', url_file)
-      click_button('Upload')
+  it 'creates the URLs' do
+    do_bulk_upload
+
+    urls.each do |url|
+      expect(SearchgovUrl.find_by(url: url)).not_to be_blank
     end
   end
 
+  it 're-indexes the domains for the URLs' do
+    do_bulk_upload
+    expect(@reindexed_domains).to eq(searchgov_domains)
+  end
+end
+
+shared_examples 'a failed bulk upload with error' do |error_message|
+  it 'sends us back to the bulk upload page' do
+    do_bulk_upload
+    expect(page).to have_text('Bulk Search.gov URL Upload')
+  end
+
+  it 'shows an error message' do
+    do_bulk_upload
+    expect(page).to have_text(error_message)
+  end
+end
+
+def do_bulk_upload
+  perform_enqueued_jobs do
+    visit url
+    attach_file('bulk_upload_urls', upload_file)
+    click_button('Upload')
+  end
+end
```

Comment:
This can be the subject of the specs:
```ruby
describe 'Bulk URL upload' do
  include ActiveJob::TestHelper
  
  subject(:do_bulk_upload) do
    perform_enqueued_jobs do
      visit url
      attach_file('bulk_upload_urls', upload_file)
      click_button('Upload')
    end
  end
```

Reply:
Changed.

Reply:
Changed.